### DISCUSSION TITLE
What new features would you find most useful?

### DISCUSSION THREAD

**Discussion by tventimi** [pulibrary/ExcelAlmaLookup#19] [Ideas]
We are looking for user insights into what features to work on next for this tool.  Please use this forum to suggest any ideas you would find useful.  Below are a few possibilities, but of course feel free to share others.  Thanks!

- A version for Mac or Google Sheets
- More general support for non-Alma sources (e.g. any catalog that supports SRU or Z39.50)
- The ability to generate a report based on search parameters rather than just a list of identifiers (e.g. "Find all records by author X with subject heading Y") 
- The ability to retrieve more item-level fields (see https://developers.exlibrisgroup.com/alma/integrations/sru/iso20775/ for a complete list of item-level fields that are visible via SRU)
- Support for other record formats (e.g. Dublin Core)
- The ability to retrieve data via the REST APIs (which would require an API key but would make it possible to extract information about acquisitions, fulfillment, etc.)

**Comment by likevel** [comment]
It would be great to have the ability to look up WorldCat holdings for specific institutions by specifying the institution symbol(s).

**Reply by tventimi** [reply]
Thanks for the suggestion!  Do you mean that you would want a yes/no response as to whether a certain item is held by a given institution?  Or would you want to retrieve other local fields specific to that institution?

**Reply by likevel** [reply]
Just a yes/no would be enough. It would be great if it could run for a (small) set of specific institutions at once.
Knowing whether a title is held elsewhere in our consortium would help us with decision making around deselections.

**Reply by tventimi** [reply]
OK, I will look into how best to implement that.  One idea that might help in the short term - the WorldCat holdings codes are taken from field 948$c, so you could enter this as a result type with a filter.  For example, the result type "948$c#PUL" would output "PUL" if that code is found in the holdings, and FALSE otherwise.  You should be able to do that with any OCLC institution code.  But you would have to create one result type per institution.

Also, I am curious, what consortium are you a part of?  If it has its own union catalog, and if that catalog is on the ReShare platform (or supports SRU or Z39.50), it might be fairly easy to add it as a "non-Alma source".  (If this wouldn't be helpful to you, then don't worry about it - I am just interested in seeing how easily the current functions of the plugin can be extended to new sources).

**Reply by tventimi** [reply]
Hello,

I just wanted to let you know that a new version of Excel Alma Lookup (v1.4.0) was released today, which allows including the OCLC holdings code as a search parameter for WorldCat.  I don't know if your use case still exists, but if so, please let me know if it addresses it and if you have any other suggestions.  Thanks!

Tom

**Comment by jacquiesamples** [comment]
I think it would be great to be able to retrieve MARC data when it contains specific indicator values.  For example, we have used local indicator values in the past when processing a reclassification project, using the MARC 050 99.  Although that project is finished, we do still use those data for troubleshooting.  Running barcode-based search to output these types of local data is very useful.

**Reply by tventimi** [reply]
Hi Jacquie,

Thanks for your message. Do you mean that you want to use indicators as a condition for retrieving a certain field (e.g. "retrieve MARC 050 only when ind1 = 9 and ind2 = 9")?

One short-term (albeit imperfect) solution would be to check the box "include indicators, subfield codes, and linkage in results for full fields", and then apply a filter to the field type, e.g. "050#99".  This would retrieve the 050 only when the result string (including indicators) contains "99".  Of course, this would also retrieve cases where "99" is found within the call number itself.

Perhaps I can add the ability to do a left-anchored filter, so that one could retrieve a certain field only when it begins with a certain string.  This would  make it possible to filter on the indicators specifically.  But please confirm first that I understand your use case correctly.  And thanks for the suggestion!

Tom Ventimiglia
Princeton University Library

**Comment by jacquiesamples** [comment]
Hi Tom,
Yes, that is what I meant. You articulated my use case much better than I did.  😊
I do know about checking the box to include indicators, etc., but I wasn’t aware of using the “#” sign as a filter.  Thanks!

Thanks for considering adding the left-anchored filter, that would be excellent!

Best,
Jacquie

**Reply by tventimi** [reply]
Hi Jacquie,

I just wanted to let you know that a new version of the Excel Alma Lookup tool (v1.4.0) was released today, which I think would address the use case you posted last year.  Results type can now be filtered using regular expressions.  So, if you check the "include indicators..." option, and include a result type of "050#/^99/", this will filter that column to only include 050s that begin with "99" (i.e. both indicators are set to 9).  If your use case still exists, please let me know if you have any other questions or suggestions.  Thanks!

Tom

**Comment by kerrieAC** [comment]
It would be great if title keyword searching could behave like true title keyword searching and not "treat the search term as a phrase and enclosed in quotes, even if it contains multiple words."
eg: searching for "Professional Psychology: Research and Practice 2017" would find display TRUE for this edition of the title but FALSE for another book of the same title but different year

**Reply by tventimi** [reply]
Hi Kerrie,

Thanks for this suggestion! I can see how this would be very useful in general.  I will open a ticket about it.  Perhaps I could just add an additional search type so users can decide whether or not to treat the search term as a phrase.  Actually, in SRU there are many indexes that support different relation types (e.g. any term, all terms, phrase), so I might also look into a way to support these in general. Right now, everything is being treated as a phrase.

Tom

**Comment by carwilcox** [comment]
I saw a great presentation on this today and am really excited about this tool! Also, this may be a training issue, not a few feature.

I would love to use this for a particular case concerning course reserves. Sometimes I need the MMS ID (and potentially other information) for records in only print or only electronic format using an ISBN. I tried filtering with the MARC 300 field containing "online" to get only electronic MMS IDs, but I still also get print MMS IDs as well. It would be great to be able to have any filters in one field limit the results of the subsequent fields as well. I'm attaching screenshots of the results I got, and the results I was hoping for.

![hoped for filtering](https://github.com/user-attachments/assets/5912c314-d784-4b2d-9a3a-0220ba65c964)
![current](https://github.com/user-attachments/assets/233e6598-5d90-430f-897c-e39831e65ded)


Caroline

**Reply by tventimi** [reply]
Hello Caroline,

For now, filters only apply to the field where they are specified, but I see the value of the functionality you describe, where an entire record is included/excluded based on the filtered field.  I'll make a ticket about this and think about the best way to implement it.  Thanks for the great suggestion!

Tom

**Reply by tventimi** [reply]
Hello Caroline,

A new version of Excel Alma Lookup (v1.4.0) was released today, which I think may address your use case.  The new multi-index search feature essentially allows you to set search filters.  For example, a search with these two parameters would search for the ISBNs in column A and only include electronic resources (i.e. those with the first character of 007 set to "c").  

<img width="470" height="73" alt="image" src="https://github.com/user-attachments/assets/3c40a322-a85a-4fa4-b2e8-14e5913b065e" />

Setting the "relation" of the second parameter to "<>" or "empty" would search for the inverse case (i.e. print resources).  I am not sure if these is a way to search the 300$a directly, but there may be other fields that indicate print vs. electronic that you can search.

I know it has been several months since your original post, but please let me know if you have any other questions or suggestions regarding this issue.  Thanks!

Tom

**Reply by carwilcox** [reply]
Hi Tom,

This works really well for my purposes. Thanks so much for adding this feature.

I use the lookup tool constantly and love it!

Cheers,

Caroline

**Comment by HXyL229ZI0gk5u** [comment]
It would be great if Excel Alma Lookup had the ability to search records in Alma using OCLC numbers in 035$a. That would allow me to use daily record merge reports from WorldShare to query records in Alma based on old OCLC numbers and update them with current OCLC numbers.

Yuji Tosaka
Cataloging/Metadata Librarian
The College of New Jersey

**Reply by tventimi** [reply]
Hello,

Thanks for your message.  I think it should be possible to search OCLC numbers using the current version of the plugin.  In the main dialog, click the "Additional Fields..." button.  This displays a full list of searchable fields.  If you type "oclc" in the "Filter" field, you will see a few search index codes for OCLC Nos.  I think the one you want is "alma.oclc_control_number_035_a", but there are other options as well.  Double-click that code and it will automatically appear in the "Field to Search" field.  Please let me know if that works for you.

Tom Ventimiglia
Princeton University Library

**Reply by HXyL229ZI0gk5u** [reply]
Awesome! Thank you for directing me to additional search fields. That has addressed my question.

**Comment by librarypunk** [comment]
I would love to use this to extract the Inventory numbers from Alma. I know I can search by Inventory number, but if I'm searching by Barcode, I want to find the corresponding Inventory number if there is one. Right now, my workaround is using Analytics or making a record set, but then I have 2 spreadsheets that I have to merge and match up, especially since I want to keep the barcodes in the order I originally put them in.

**Reply by tventimi** [reply]
Thanks for your suggestion - it's actually a feature that others have requested before!  Unfortunately (and surprisingly) inventory numbers are not included in the SRU output, even though they are searchable.  So, it is not possible to add this feature at this time, but if the SRU output ever changes, we will be sure to make this field available.

Tom Ventimiglia
Princeton University Library

**Comment by carwilcox** [comment]
Hi Tom,

I would love to be able to pull more information from electronic portfolios, especially Access Type, Public Access Model, PO Line and possibly Internal Description and/or General Notes.

Thanks for considering!

Caroline

> We are looking for user insights into what features to work on next for this tool. Please use this forum to suggest any ideas you would find useful. Below are a few possibilities, but of course feel free to share others. Thanks!

> * The ability to retrieve more item-level fields (see https://developers.exlibrisgroup.com/alma/integrations/sru/iso20775/ for a complete list of item-level fields that are visible via SRU)

**Reply by tventimi** [reply]
Hi Caroline,

When you get a chance, can you please send me a few links to records at your institution that have the fields in question?  I realize these fields may not be visible in your public catalog, but even just having the MMS IDs would be helpful.  This way, I can identify which ISO Holdings elements (if any) the fields correspond to.  Unfortunately, I don't believe that PO Line Number is a retrievable field (though it should be available as a search index.)  I see in the ISO Holdings documentation there are fields for notes and descriptions, but I am not sure how these correspond to the fields in the portfolios, which is why I wanted to check some examples. Thanks!

Tom

**Comment by carwilcox** [comment]
Hi Tom,

Sure, here are a couple MMS IDs: 

- [Advanced practice nursing procedures](https://librarysearch.ohsu.edu/permalink/01ALLIANCE_OHSU/19jn9i0/alma99900593318301858)  - Access Type = perpetual, Public Access Model = Limited user access (3 copies available), PO Line = POL-9022
- [Maternal-newborn nursing](https://librarysearch.ohsu.edu/permalink/01ALLIANCE_OHSU/19jn9i0/alma99900554852101858) - Access Type = blank, Public Access Model = Open Access, PO Line = blank
- [Lashley's essentials of clinical genetics in nursing practice](https://librarysearch.ohsu.edu/permalink/01ALLIANCE_OHSU/19jn9i0/alma99900084077901858) - this one has two portfolios attached to it so hopefully you'll get results for both

Thanks for looking into this!

Cheers,

Caroline

**Reply by tventimi** [reply]
Hi Caroline,

Thank you for providing these examples.  I looked into them more, and unfortunately, it seems that the SRU integration provides limited metadata about electronic portfolios.  I could not find any fields in the SRU output for these records corresponding to access type, public access model, or PO line.  I did find (by looking at a few Princeton records), that the "public note" field of the portfolio can be retrieved by entering "AVE$n" as a result type.  However, I do not see a way to retrieve other types of notes, including the internal description.

I'm sorry I could not be more help with this, but feel free to ask if there are any other types of data you are interested in. 

Tom
